File: CS/DXRichEditControlAPISample/CodeExamples/PageLayoutActions.cs
using DevExpress.XtraRichEdit.API.Native;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DXRichEditControlAPISample.CodeExamples
{
    class PageLayoutActions
    {
        static void LineNumbering(Document document)
        {
            #region #LineNumbering
            document.LoadDocument("Documents//Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml);
            document.Unit = DevExpress.Office.DocumentUnit.Inch;
            Section sec = document.Sections[0];
            sec.LineNumbering.CountBy = 2;
            sec.LineNumbering.Start = 1;
            sec.LineNumbering.Distance = 0.25f;
            sec.LineNumbering.RestartType = LineNumberingRestart.NewSection;
            #endregion #LineNumbering
        }

        static void CreateColumns(Document document)
        {
            #region #CreateColumns
            document.LoadDocument("Documents//Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml);
            document.Unit = DevExpress.Office.DocumentUnit.Inch;
            // Get the first section in a document
            Section firstSection = document.Sections[0];
            // Create columns and apply them to the document
            SectionColumnCollection sectionColumnsLayout =
                firstSection.Columns.CreateUniformColumns(firstSection.Page, 0.2f, 3);
            firstSection.Columns.SetColumns(sectionColumnsLayout);
            #endregion #CreateColumns
        }
        
        static void PrintLayout(Document document)
        {
            #region #PrintLayout
            document.Unit = DevExpress.Office.DocumentUnit.Inch;
            document.Sections[0].Page.PaperKind = DevExpress.Drawing.Printing.DXPaperKind.A6;
            document.Sections[0].Page.Landscape = true;
            document.Sections[0].Margins.Left = 2.0f;
            #endregion #PrintLayout
        }

        static void TabStops(Document document)
        {
            #region #TabStops
            document.Unit = DevExpress.Office.DocumentUnit.Inch;
            TabInfoCollection tabs = document.Paragraphs[0].BeginUpdateTabs(true);
            DevExpress.XtraRichEdit.API.Native.TabInfo tab1 = new DevExpress.XtraRichEdit.API.Native.TabInfo();
            // Sets tab stop at 2.5 inch
            tab1.Position = 2.5f;
            tab1.Alignment = TabAlignmentType.Left;
            tab1.Leader = TabLeaderType.MiddleDots;
            tabs.Add(tab1);
            DevExpress.XtraRichEdit.API.Native.TabInfo tab2 = new DevExpress.XtraRichEdit.API.Native.TabInfo();
            tab2.Position = 5.5f;
            tab2.Alignment = TabAlignmentType.Decimal;
            tab2.Leader = TabLeaderType.EqualSign;
            tabs.Add(tab2);
            document.Paragraphs[0].EndUpdateTabs(tabs);
            #endregion #TabStops
        }
        static void CreatePageBorders(Document document)
        {
            #region #CreatePageBorders
            // Generate a document with two sections and multiple pages in each section.
            document.AppendText("\f\f\f");
            document.Paragraphs.Append();
            document.AppendSection();
            document.AppendText("\f\f");

            Section firstSection = document.Sections[0];
            SectionPageBorders pageBorders1 = firstSection.PageBorders;

            // Set page borders for the first page of the first section.
            PageBorderHelper.SetPageBorders(pageBorders1.LeftBorder, BorderLineStyle.Single, 1f, System.Drawing.Color.Red);
            PageBorderHelper.SetPageBorders(pageBorders1.TopBorder, BorderLineStyle.Single, 1f, System.Drawing.Color.Red);
            PageBorderHelper.SetPageBorders(pageBorders1.RightBorder, BorderLineStyle.Single, 1f, System.Drawing.Color.Red);
            PageBorderHelper.SetPageBorders(pageBorders1.BottomBorder, BorderLineStyle.Single, 1f, System.Drawing.Color.Red);
            pageBorders1.AppliesTo = PageBorderAppliesTo.FirstPage;

            Section secondSection = document.Sections[1];
            SectionPageBorders pageBorders2 = secondSection.PageBorders;

            // Set page borders for all pages of the second section.
            PageBorderHelper.SetPageBorders(pageBorders2.LeftBorder, BorderLineStyle.Double, 1.5f, System.Drawing.Color.Green);
            PageBorderHelper.SetPageBorders(pageBorders2.TopBorder, BorderLineStyle.Double, 1.5f, System.Drawing.Color.Green);
            PageBorderHelper.SetPageBorders(pageBorders2.RightBorder, BorderLineStyle.Double, 1.5f, System.Drawing.Color.Green);
            PageBorderHelper.SetPageBorders(pageBorders2.BottomBorder, BorderLineStyle.Double, 1.5f, System.Drawing.Color.Green);
            pageBorders2.AppliesTo = PageBorderAppliesTo.AllPages;
            pageBorders2.ZOrder = PageBorderZOrder.Back;
            #endregion #CreatePageBorders

        }
        #region #@CreatePageBorders
        class PageBorderHelper
        {
            public static void SetPageBorders(PageBorder border, BorderLineStyle lineStyle,
            float borderWidth, System.Drawing.Color color)
            {
                border.LineStyle = lineStyle;
                border.LineWidth = borderWidth;
                border.LineColor = color;
            }
        }
        #endregion #@CreatePageBorders
    }
}

File: VB/DXRichEditControlAPISample/CodeExamples/PageLayoutActions.vb
Imports DevExpress.XtraRichEdit.API.Native
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace DXRichEditControlAPISample.CodeExamples

    Friend Class PageLayoutActions

        Private Shared Sub LineNumbering(ByVal document As DevExpress.XtraRichEdit.API.Native.Document)
#Region "#LineNumbering"
            document.LoadDocument("Documents//Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            document.Unit = DevExpress.Office.DocumentUnit.Inch
            Dim sec As DevExpress.XtraRichEdit.API.Native.Section = document.Sections(0)
            sec.LineNumbering.CountBy = 2
            sec.LineNumbering.Start = 1
            sec.LineNumbering.Distance = 0.25F
            sec.LineNumbering.RestartType = DevExpress.XtraRichEdit.API.Native.LineNumberingRestart.NewSection
#End Region  ' #LineNumbering
        End Sub

        Private Shared Sub CreateColumns(ByVal document As DevExpress.XtraRichEdit.API.Native.Document)
#Region "#CreateColumns"
            document.LoadDocument("Documents//Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            document.Unit = DevExpress.Office.DocumentUnit.Inch
            ' Get the first section in a document
            Dim firstSection As DevExpress.XtraRichEdit.API.Native.Section = document.Sections(0)
            ' Create columns and apply them to the document
            Dim sectionColumnsLayout As DevExpress.XtraRichEdit.API.Native.SectionColumnCollection = firstSection.Columns.CreateUniformColumns(firstSection.Page, 0.2F, 3)
            firstSection.Columns.SetColumns(sectionColumnsLayout)
#End Region  ' #CreateColumns
        End Sub

        Private Shared Sub PrintLayout(ByVal document As DevExpress.XtraRichEdit.API.Native.Document)
#Region "#PrintLayout"
            document.Unit = DevExpress.Office.DocumentUnit.Inch
            document.Sections(CInt((0))).Page.PaperKind = DevExpress.Drawing.Printing.DXPaperKind.A6
            document.Sections(CInt((0))).Page.Landscape = True
            document.Sections(CInt((0))).Margins.Left = 2.0F
#End Region  ' #PrintLayout
        End Sub

        Private Shared Sub TabStops(ByVal document As DevExpress.XtraRichEdit.API.Native.Document)
#Region "#TabStops"
            document.Unit = DevExpress.Office.DocumentUnit.Inch
            Dim tabs As DevExpress.XtraRichEdit.API.Native.TabInfoCollection = document.Paragraphs(CInt((0))).BeginUpdateTabs(True)
            Dim tab1 As DevExpress.XtraRichEdit.API.Native.TabInfo = New DevExpress.XtraRichEdit.API.Native.TabInfo()
            ' Sets tab stop at 2.5 inch
            tab1.Position = 2.5F
            tab1.Alignment = DevExpress.XtraRichEdit.API.Native.TabAlignmentType.Left
            tab1.Leader = DevExpress.XtraRichEdit.API.Native.TabLeaderType.MiddleDots
            tabs.Add(tab1)
            Dim tab2 As DevExpress.XtraRichEdit.API.Native.TabInfo = New DevExpress.XtraRichEdit.API.Native.TabInfo()
            tab2.Position = 5.5F
            tab2.Alignment = DevExpress.XtraRichEdit.API.Native.TabAlignmentType.[Decimal]
            tab2.Leader = DevExpress.XtraRichEdit.API.Native.TabLeaderType.EqualSign
            tabs.Add(tab2)
            document.Paragraphs(CInt((0))).EndUpdateTabs(tabs)
#End Region  ' #TabStops
        End Sub

        Private Shared Sub CreatePageBorders(ByVal document As DevExpress.XtraRichEdit.API.Native.Document)
#Region "#CreatePageBorders"
            ' Generate a document with two sections and multiple pages in each section.
            document.AppendText(Global.Microsoft.VisualBasic.Constants.vbFormFeed & Global.Microsoft.VisualBasic.Constants.vbFormFeed & Global.Microsoft.VisualBasic.Constants.vbFormFeed)
            document.Paragraphs.Append()
            document.AppendSection()
            document.AppendText(Global.Microsoft.VisualBasic.Constants.vbFormFeed & Global.Microsoft.VisualBasic.Constants.vbFormFeed)
            Dim firstSection As DevExpress.XtraRichEdit.API.Native.Section = document.Sections(0)
            Dim pageBorders1 As DevExpress.XtraRichEdit.API.Native.SectionPageBorders = firstSection.PageBorders
            ' Set page borders for the first page of the first section.
            Call DXRichEditControlAPISample.CodeExamples.PageLayoutActions.PageBorderHelper.SetPageBorders(pageBorders1.LeftBorder, DevExpress.XtraRichEdit.API.Native.BorderLineStyle.[Single], 1F, System.Drawing.Color.Red)
            Call DXRichEditControlAPISample.CodeExamples.PageLayoutActions.PageBorderHelper.SetPageBorders(pageBorders1.TopBorder, DevExpress.XtraRichEdit.API.Native.BorderLineStyle.[Single], 1F, System.Drawing.Color.Red)
            Call DXRichEditControlAPISample.CodeExamples.PageLayoutActions.PageBorderHelper.SetPageBorders(pageBorders1.RightBorder, DevExpress.XtraRichEdit.API.Native.BorderLineStyle.[Single], 1F, System.Drawing.Color.Red)
            Call DXRichEditControlAPISample.CodeExamples.PageLayoutActions.PageBorderHelper.SetPageBorders(pageBorders1.BottomBorder, DevExpress.XtraRichEdit.API.Native.BorderLineStyle.[Single], 1F, System.Drawing.Color.Red)
            pageBorders1.AppliesTo = DevExpress.XtraRichEdit.API.Native.PageBorderAppliesTo.FirstPage
            Dim secondSection As DevExpress.XtraRichEdit.API.Native.Section = document.Sections(1)
            Dim pageBorders2 As DevExpress.XtraRichEdit.API.Native.SectionPageBorders = secondSection.PageBorders
            ' Set page borders for all pages of the second section.
            Call DXRichEditControlAPISample.CodeExamples.PageLayoutActions.PageBorderHelper.SetPageBorders(pageBorders2.LeftBorder, DevExpress.XtraRichEdit.API.Native.BorderLineStyle.[Double], 1.5F, System.Drawing.Color.Green)
            Call DXRichEditControlAPISample.CodeExamples.PageLayoutActions.PageBorderHelper.SetPageBorders(pageBorders2.TopBorder, DevExpress.XtraRichEdit.API.Native.BorderLineStyle.[Double], 1.5F, System.Drawing.Color.Green)
            Call DXRichEditControlAPISample.CodeExamples.PageLayoutActions.PageBorderHelper.SetPageBorders(pageBorders2.RightBorder, DevExpress.XtraRichEdit.API.Native.BorderLineStyle.[Double], 1.5F, System.Drawing.Color.Green)
            Call DXRichEditControlAPISample.CodeExamples.PageLayoutActions.PageBorderHelper.SetPageBorders(pageBorders2.BottomBorder, DevExpress.XtraRichEdit.API.Native.BorderLineStyle.[Double], 1.5F, System.Drawing.Color.Green)
            pageBorders2.AppliesTo = DevExpress.XtraRichEdit.API.Native.PageBorderAppliesTo.AllPages
            pageBorders2.ZOrder = DevExpress.XtraRichEdit.API.Native.PageBorderZOrder.Back
#End Region  ' #CreatePageBorders
        End Sub

#Region "#@CreatePageBorders"
        Private Class PageBorderHelper

            Public Shared Sub SetPageBorders(ByVal border As DevExpress.XtraRichEdit.API.Native.PageBorder, ByVal lineStyle As DevExpress.XtraRichEdit.API.Native.BorderLineStyle, ByVal borderWidth As Single, ByVal color As System.Drawing.Color)
                border.LineStyle = lineStyle
                border.LineWidth = borderWidth
                border.LineColor = color
            End Sub
        End Class
#End Region  ' #@CreatePageBorders
    End Class
End Namespace
